From bee9dd659578a34b23d5db5e3dcbcbcd4ce6fab6 Mon Sep 17 00:00:00 2001
From: Timo Aaltonen <timo.aaltonen@canonical.com>
Date: Mon, 11 Sep 2017 14:50:07 +0300
Subject: [PATCH 1/2] Revert "mm: replace get_user_pages() write/force
 parameters with gup_flags"

This reverts commit 768ae309a96103ed02eb1e111e838c87854d8b51.
---
 arch/cris/arch-v32/drivers/cryptocop.c                 |  4 +++-
 arch/ia64/kernel/err_inject.c                          |  2 +-
 arch/x86/mm/mpx.c                                      |  5 +++--
 drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c                |  7 ++-----
 drivers/gpu/drm/radeon/radeon_ttm.c                    |  3 +--
 drivers/gpu/drm/via/via_dmablit.c                      |  4 ++--
 drivers/infiniband/core/umem.c                         |  6 +-----
 drivers/infiniband/hw/mthca/mthca_memfree.c            |  2 +-
 drivers/infiniband/hw/qib/qib_user_pages.c             |  3 +--
 drivers/infiniband/hw/usnic/usnic_uiom.c               |  5 +----
 drivers/media/v4l2-core/videobuf-dma-sg.c              |  7 ++-----
 drivers/misc/mic/scif/scif_rma.c                       |  3 ++-
 drivers/misc/sgi-gru/grufault.c                        |  2 +-
 drivers/platform/goldfish/goldfish_pipe.c              |  3 +--
 drivers/rapidio/devices/rio_mport_cdev.c               |  3 +--
 .../vc04_services/interface/vchiq_arm/vchiq_2835_arm.c |  3 ++-
 .../vc04_services/interface/vchiq_arm/vchiq_arm.c      |  3 ++-
 drivers/virt/fsl_hypervisor.c                          |  4 ++--
 include/linux/mm.h                                     |  2 +-
 mm/gup.c                                               | 12 +++++++++---
 mm/mempolicy.c                                         |  2 +-
 mm/nommu.c                                             | 18 ++++++++++++++----
 22 files changed, 54 insertions(+), 49 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c
index 264899d..ee56544 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c
@@ -555,13 +555,10 @@ struct amdgpu_ttm_tt {
 int amdgpu_ttm_tt_get_user_pages(struct ttm_tt *ttm, struct page **pages)
 {
 	struct amdgpu_ttm_tt *gtt = (void *)ttm;
-	unsigned int flags = 0;
+	int write = !(gtt->userflags & AMDGPU_GEM_USERPTR_READONLY);
 	unsigned pinned = 0;
 	int r;
 
-	if (!(gtt->userflags & AMDGPU_GEM_USERPTR_READONLY))
-		flags |= FOLL_WRITE;
-
 	if (gtt->userflags & AMDGPU_GEM_USERPTR_ANONONLY) {
 		/* check that we only use anonymous memory
 		   to prevent problems with writeback */
@@ -584,7 +581,7 @@ int amdgpu_ttm_tt_get_user_pages(struct ttm_tt *ttm, struct page **pages)
 		list_add(&guptask.list, &gtt->guptasks);
 		spin_unlock(&gtt->guptasklock);
 
-		r = get_user_pages(userptr, num_pages, flags, p, NULL);
+		r = get_user_pages(userptr, num_pages, write, 0, p, NULL);
 
 		spin_lock(&gtt->guptasklock);
 		list_del(&guptask.list);
diff --git a/drivers/gpu/drm/radeon/radeon_ttm.c b/drivers/gpu/drm/radeon/radeon_ttm.c
index 4ce04e0..776e798 100644
--- a/drivers/gpu/drm/radeon/radeon_ttm.c
+++ b/drivers/gpu/drm/radeon/radeon_ttm.c
@@ -566,8 +566,7 @@ static int radeon_ttm_tt_pin_userptr(struct ttm_tt *ttm)
 		uint64_t userptr = gtt->userptr + pinned * PAGE_SIZE;
 		struct page **pages = ttm->pages + pinned;
 
-		r = get_user_pages(userptr, num_pages, write ? FOLL_WRITE : 0,
-				   pages, NULL);
+		r = get_user_pages(userptr, num_pages, write, 0, pages, NULL);
 		if (r < 0)
 			goto release_pages;
 
-- 
2.7.4

